### ISSUE TITLE
Azure Data Studio crashes after selecting 'collapse cells' in existing SQL notebook

### ISSUE THREAD

**Issue by Catnary** [microsoft/azuredatastudio#22942]
Issue Type: <b>Bug</b>

Clicking the 'collapse cells' icon in an existing SQL notebook causes Azure Data Studio to hang. Eventually, a *Window is not responding* popup appears. 

After selecting 'Keep waiting' from the list of options in the popup, functionality is sometimes restored. Other times the application crashes completely. Message reads 

> The window has crashed (reason: 'crashed', code:'-36861')

I have VSCode 1.77.3 installed on the same machine, but not running when this problem occurs.

Azure Data Studio version: azuredatastudio 1.43.0 (b790d700898b1095d83e62f0de14678a58222520, 2023-04-10T23:14:27.236Z)
OS version: Windows_NT x64 10.0.19042
Restricted Mode: No
Preview Features: Enabled

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|11th Gen Intel(R) Core(TM) i5-1145G7 @ 2.60GHz (8 x 2611)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|undefined|
|Memory (System)|15.69GB (2.87GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (12)</summary>

Extension|Author (truncated)|Version
---|---|---
admin-pack|Mic|0.0.2
agent|Mic|0.49.0
azcli|Mic|1.8.0
dacpac|Mic|1.13.0
import|Mic|1.5.4
powershell|ms-|2023.2.1
profiler|Mic|0.12.2
query-history|Mic|0.5.3
schema-compare|Mic|1.18.0
sql-database-projects|Mic|1.0.1
sql-prompt|Red|0.2.12
sql-search|Red|0.4.2

</details>

**Comment by chlafreniere** [comment]
@corivera can you take a look please

**Comment by corivera** [comment]
Sure

**Comment by corivera** [comment]
I'm not able to repro this in insiders or stable. @Catnary does this happen with every notebook for you, or only specific ones? If the latter, would you be be able to share that notebook?

**Comment by Catnary** [comment]
@corivera Thanks for looking into this for me. I can't share the notebook unfortunately, but can give some more details: I have seen it happen with several different notebooks previously. However, I just opened a sample of three different SQL notebooks and tried to reproduce the issue, and found it only seems to happen with one of them. 

That particular notebook has some relatively long queries (100+ lines per cell) and some of those return 5k+ rows. If I clear the results, collapse cells appears to work ok. If I re-run the cells, it causes the problem described above. So the issue may be related to having large datasets in the notebook.

I was actually in the process of re-writing those queries to return more notebook-friendly results, so this should be easy to avoid.

As a separate issue, I have found that if I do get the not responding popup and then select the reopen option, when it reopens, the kernel dropdown gets stuck on 'Loading kernels...' and the Attach to dropdown shows 'Loading contexts...'. I have to manually close and restart ADS for it to find the SQL kernel and again. I'll check to see if that's already been reported and add a new ticket if not.
